File: training-data/java/Palindrome.java
public class Palindrome {
    public static boolean isPalindrome(String s) {
        String cleaned = s.toLowerCase().replaceAll("[^a-z0-9]", "");
        int left = 0;
        int right = cleaned.length() - 1;
        while (left < right) {
            if (cleaned.charAt(left) != cleaned.charAt(right)) return false;
            left++;
            right--;
        }
        return true;
    }

    public static void main(String[] args) {
        System.out.println(isPalindrome("A man a plan a canal Panama"));
        System.out.println(isPalindrome("hello"));
    }
}

File: training-data/php/palindrome.php
<?php

function isPalindrome(string $s): bool
{
    $cleaned = preg_replace('/[^a-z0-9]/', '', strtolower($s));
    return $cleaned === strrev($cleaned);
}

var_dump(isPalindrome("A man a plan a canal Panama"));
var_dump(isPalindrome("hello"));
